### DISCUSSION TITLE
Workflow uniqueness

### DISCUSSION THREAD

**Discussion by lyuboxa** [cschleiden/go-workflows#259] [Q&A] [answered]
Was the intended purpose for workflow instances with the same ID uniqueness not to be enforced?
Currently the PK is on both `instance_id` and `execution_id` which allows for two subsequent `CreateWorkflowInstance` 
calls to succeed. 

Referring to how this is handled in Cadence and Temporal:

* Cadence https://cadenceworkflow.io/docs/concepts/workflows/#id-uniqueness
* In Temporal, the ID is unique within a namespace https://docs.temporal.io/workflows#workflow-id

Thanks!

**Comment by cschleiden** [comment, marked as answer]
No, that's a good point, that should change. Creating an issue from this.
